thriftclient/client_test.go

```diff
@@ -0,0 +1,172 @@
+package thriftclient_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/reddit/baseplate.go/mqsend"
+	"github.com/reddit/baseplate.go/runtimebp"
+	"github.com/reddit/baseplate.go/thriftclient"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+const (
+	method = "foo"
+)
+
+func initClients() (*thriftclient.MockClient, *thriftclient.RecordedClient, *thriftclient.MonitoredClient) {
+	mock := &thriftclient.MockClient{}
+	recorder := thriftclient.NewRecordedClient(mock)
+	client := thriftclient.NewMonitoredClient(recorder)
+	return mock, recorder, client
+}
+
+func initServerSpan(ctx context.Context) (context.Context, *tracing.Tracer) {
+	ip, _ := runtimebp.GetFirstIPv4()
+	tracer := &tracing.Tracer{
+		SampleRate: 1.0,
+		Endpoint: tracing.ZipkinEndpointInfo{
+			ServiceName: "test-service",
+			IPv4:        ip,
+		},
+	}
+	tracer.Recorder = mqsend.OpenMockMessageQueue(mqsend.MessageQueueConfig{
+		MaxQueueSize:   100,
+		MaxMessageSize: 1024,
+	})
+
+	ctx, span := tracing.CreateServerSpanForContext(ctx, tracer, "test-service")
+	span.SetDebug(true)
+	return ctx, tracer
+}
+
+func initLocalSpan(ctx context.Context) (context.Context, *tracing.Tracer) {
+	ctx, tracer := initServerSpan(ctx)
+	span := tracing.GetServerSpan(ctx)
+	if span == nil {
+		panic("server span was nill")
+	}
+	ctx, _ = span.CreateLocalChildForContext(ctx, "local-test", "")
+	return ctx, tracer
+}
+
+func drainTracer(t *tracing.Tracer) ([]byte, error) {
+	mmq := t.Recorder.(*mqsend.MockMessageQueue)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+	defer cancel()
+	return mmq.Receive(ctx)
+}
+
+func TestMonitoredClientNilSpan(t *testing.T) {
+	t.Parallel()
+
+	_, recorder, client := initClients()
+	client.Call(context.Background(), method, nil, nil)
```

Comment:
Should we check the error here?

Reply:
It's not really necessary _as this code is written_ but it's probably a good idea since this is relying on the Mock behavior not changing so that would provide a nice guard to that.